### PR TITLE
Fix transaction level capability check

### PR DESCRIPTION
**Pull request by diosmosis**

### CODE REVIEW

File: core/Db/TransactionLevel.php [hunk 1]
```diff
@@ -37,12 +40,15 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        $backup = $this->db->fetchOne('SELECT @@TX_ISOLATION');
```
Comment (tsteur) [inline core/Db/TransactionLevel.php]: @diosmosis does this need to be `$this->statusBackup` instead of `$backup`?

Comment (diosmosis) [reply core/Db/TransactionLevel.php]: That's done below after successfully changing it. If it fails for some reason to set it then we don't need to set it back.

Comment (tsteur) [reply core/Db/TransactionLevel.php]: makes sense 👍

**Comment by tsteur** [comment]
@diosmosis fyi I'm now getting an error 

`SQLSTATE[HY000]: General error: 1193 Unknown system variable 'TX_ISOLATION' - in plugin Goals` see https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-3.html

with MySQL8. Will investigate and create PR.

**Pull request merged**